emm/src/main/java/whelk/EmmChangeSet.java

```diff
@@ -0,0 +1,143 @@
+package whelk;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.sql.*;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import static whelk.util.Jackson.mapper;
+
+public class EmmChangeSet {
+    public static final int TARGET_HITS_PER_PAGE = 100;
+    private static final Logger logger = LogManager.getLogger(EmmChangeSet.class);
+
+    static void sendChangeSet(Whelk whelk, HttpServletResponse res, String until, String apiBaseUrl) throws IOException {
+
+        List<EmmActivity> activitiesOnPage = new ArrayList<>(TARGET_HITS_PER_PAGE+5);
+        Timestamp nextTimeStamp = getPage(whelk, until, activitiesOnPage);
+        if (nextTimeStamp == null) {
+            res.sendError(500);
+            return;
+        }
+
+        // THIS SHIT is so painful in Java :(
```

Comment:
Can use `Map.of(...)` and `List.of(...)`
`new HashMap<>(Map.of(...))` if mutability is needed.

Still pretty painful